--- thinkeramik-3.1.4/styleconf/thinkeramikconf.cpp.orig	2004-04-17 13:09:08.034674984 +0200
+++ thinkeramik-3.1.4/styleconf/thinkeramikconf.cpp	2004-04-17 13:16:30.517407384 +0200
@@ -1245,8 +1245,10 @@
 void ThinKeramikStyleConfig::slotLoadScheme()
 {
     QString schemeStr  = "/thinkeramik_" + schemeName->text() + ".kcm";
-    QString schemeFile = QDir::homeDirPath() + "/.qt" + schemeStr + "rc";
-    if (QFile::exists(schemeFile)) {
+    QStringList schemeFile( (QString) (QDir::homeDirPath() + "/.qt" + schemeStr + "rc"));
+    schemeFile << (QString) ("/usr/share/apps/kdisplay/color-schemes" + schemeStr + "rc");
+    for ( QStringList::Iterator it = schemeFile.begin(); it != schemeFile.end(); ++it ) {
+    if (QFile::exists(*it)) {
         int loadOK = QMessageBox::question(this,
 					   i18n("Load Scheme? - Thin Keramik Scheme Manager"),
 					   i18n("Do you load \"%1\"?").arg(schemeName->text()) + "\n"
@@ -1264,15 +1266,15 @@
 				     QMessageBox::NoButton);
 	    emit changed(true);
 	}
-    } else {
+/*    } else {
         QMessageBox::critical(this,
 			      i18n("Error - Thin Keramik Scheme Manager"),
 			      i18n("\"%1\" : No such scheme.").arg(schemeName->text()),
 			      QMessageBox::Cancel,
 			      QMessageBox::NoButton,
-			      QMessageBox::NoButton);          
+			      QMessageBox::NoButton);         */
     }
-
+  }
 }
 
 void ThinKeramikStyleConfig::slotSaveScheme()
@@ -1369,8 +1371,10 @@
 void ThinKeramikStyleConfig::searchScheme()
 {
     schemeListBox->clear();
-
-    QDir schemeDir(QDir::homeDirPath() + "/.qt", "thinkeramik_*.kcmrc");
+    QStringList schemeDirs (QDir::homeDirPath() + "/.qt");
+    schemeDirs << (QString) ("/usr/share/apps/kdisplay/color-schemes");
+    for ( QStringList::Iterator it = schemeDirs.begin(); it != schemeDirs.end(); ++it ) {
+    QDir schemeDir(*it, "thinkeramik_*.kcmrc");
     uint schemeCount = schemeDir.count();
     for (uint sidx = 0; sidx<schemeCount; sidx++) {
           QString schemeFile = schemeDir[sidx];
@@ -1381,7 +1385,7 @@
 		schemeListBox->insertItem(schemeFile);
 	  }
     }
-
+  }
 }
 
 #include "thinkeramikconf.moc"
